Assets/LeapMotion/Scripts/Utils/Utils.cs

```diff
@@ -430,6 +430,46 @@ public struct ChildrenEnumerator {
       capsule.height = (capsuleSpaceDistToA + capsule.radius) * 2;
     }
 
+    /// <summary>
+    /// Recursively searches the hierarchy of the argument GameObject to find all of the
+    /// Colliders that are attached to the object's Rigidbody (or that _would_ be 
+    /// attached to its Rigidbody if it doesn't have one) and adds them to the provided
+    /// colliders list. Warning: The provided "colliders" List will be cleared before
+    /// use.
+    /// 
+    /// Colliders that are the children of other Rigidbody elements beneath the argument
+    /// object are ignored.
+    /// </summary>
+    public static void FindColliders<T>(GameObject obj, ref List<T> colliders) where T : Collider {
+      colliders.Clear();
+      Stack<Transform> toVisit = Pool<Stack<Transform>>.Spawn();
```

Comment:
A good best practice I have adopted is to do something like this when using Pool

```
var resources = Pool.Spawn();
try {
  // do stuff with resource
} finally {
  resource.Clear();
  Pool.Recycle(resource)
}
```

That way you can ensure that there is never a memory leak due to unexpected errors (or returns!) within the method.